### ISSUE TITLE
whisper : update the "talk" example

### ISSUE THREAD

**Issue by ggerganov** [ggerganov/whisper.cpp#1818]
The `talk` and `talk.wasm` examples have become a bit stale, using a very old implementation of `gpt-2`:

- https://github.com/ggerganov/whisper.cpp/tree/master/examples/talk
- https://github.com/ggerganov/whisper.cpp/tree/master/examples/talk.wasm

It would be nice to bring those examples up-to-date using the latest `gpt-2` implementation from `ggml`: https://github.com/ggerganov/ggml/tree/master/examples/gpt-2

**Comment by ggerganov** [comment]
No, you have to copy the `gpt-2` implementation from the `examples/gpt-2` in the `ggml` repo to the `examples/talk` in this repo. The `whisper.cpp` file does not need any changes

**Comment by shivghai** [comment]
@ggerganov - my attempt here - https://github.com/ggerganov/whisper.cpp/pull/2384

**Comment by ggerganov** [comment]
@shivghai Thank you - will try to take a look soon

**Comment by xiaogz** [comment]
@shivghai I'm still encountering the same error message when running the talk example in your branch compared to master at `5b1ce40 2024-09-17 server-use-OS-generated-temp-file-name-for-converted-files-2419 - Toliver`. I'm using a mac studio m2 ultra on Sonoma 14.7.

Are you still working on your PR?

**Comment by shivghai** [comment]
@xiaogz the PR is up to date/ready - i just tried using my branch, and i can get it to run. Steps to run on my PR:

1. `make talk`
2. `./talk`

How are you currently building/running?

**Comment by xiaogz** [comment]
@shivghai My bad! I made a git mistake. I fetched your forked branch properly and it works.

I also tried the fix myself too which differs from yours: 
- copy over ggml's include and src at commit `e7b2390 2024-09-20`
- fix build breakage in whisper.cpp:
```
diff --git a/src/whisper.cpp b/src/whisper.cpp
index 35874aa..cf7c2c5 100644
--- a/src/whisper.cpp
+++ b/src/whisper.cpp
@@ -177,7 +177,7 @@ static bool ggml_graph_compute_helper(
                          int   n_threads,
          ggml_abort_callback   abort_callback,
                         void * abort_callback_data) {
-    struct ggml_cplan plan = ggml_graph_plan(graph, n_threads);
+    struct ggml_cplan plan = ggml_graph_plan(graph, n_threads, nullptr);

     plan.abort_callback      = abort_callback;
     plan.abort_callback_data = abort_callback_data;
```
- fix the buffer size calculations in examples/talk/gpt-2.cpp
